buildings/gui/alter_building_relationships.py

```diff
@@ -464,20 +478,64 @@ def related_clicked(self, commit_status=True):
             else:
                 self.btn_save.setEnabled(True)
 
+    @pyqtSlot()
+    def delete_clicked(self, commit_status=True):
+        self.btn_matched.setEnabled(False)
```

Comment:
if we disable the matched/related button in `reason_given` we don't need to run `self.switch_btn_match_and_related()` when canceling

Reply:
The buttons must be disabled here so that between clicking ok on the dialog and save the user can't then try relate/match. The line `self.switch_btn_match_and_related()` was added so that if the user clicked cancel or the processing failed the match/relate btns would enable again. Open to suggestions for improving the workflow though